components/root/root.tsx

```diff
@@ -597,6 +643,7 @@ export default class Root extends React.PureComponent<Props, State> {
                                         <Pluggable
                                             pluggableName={'CustomRouteComponent'}
                                             pluggableId={plugin.id}
+                                            css={{gridArea: 'center'}}
```

Comment:
This is styled-components' css prop being used as a fallback that specifies a `className` that would route the output of this pluggable to `grid-area: center`  IF the registered component has a `className` prop (or uses the `...rest` pattern).

Reply:
The `grid-template` is in `_structure.scss`

Reply:
Ah, I didn't realize how grid-templates worked when I wrote that.

On a related note, it's weird to me to see the `css` prop from styled-components contain an object. I wonder if it performs better than using `style` due to how it presumably adds a class that applies the style rather than doing it inline

Reply:
It can take an object or css string in the same way that `css` can accept either: 

```tsx
import {css} from 'styled-components';
const style: FlattenSimpleInterpolation = css({color: 'red'});
const other: FlattenSimpleInterpolation = css`
    color: red;
`;
```


Under the hood, I believe the css prop is precompiled by babel, so I think the performance is the same as with `styled` wrapping:
```tsx
const thing = styled(Pluggable)`
	color: red;
`;
```

Reply:
That seems a bit weird to me to see `styled` being used to wrap a React component, especially one that doesn't take a `style` prop. It may also result in a `className` prop being added since I don't remember which styled-components uses, but that confusion applies either way.

Not blocking, but does this require that the component provided by the plugin accepts a style prop? It seems like this might not work if the provided component doesn't because the styles wouldn't be applied to anything.

Reply:
`styled(...)`-wrapping a component, or using the `css` prop, will yield a `className` addition. Indeed, `className` or rest attrs would need to be supported in the registered `CustomRouteComponent`. (This is supported via `Pluggable`'s rest-spread props.)